File: 0217-contains-duplicate/0217-contains-duplicate.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Boolean
     */
    function containsDuplicate($nums) {
        $hashMap = array();

        for ($i = 0; $i < count($nums); $i++){
            if ($hashMap[$nums[$i]]){
                return true;
            }
            $hashMap[$nums[$i]] = true;
        }

        return false;
    }
}

File: 0217-contains-duplicate/0217-contains-duplicate.rs
use std::collections::HashMap;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut hashMap = HashMap::new();

        for n in nums{
            if (hashMap.contains_key(&n)){
                return true;
            }
            hashMap.insert(n, true);
        }

        return false;
    }
}

    // contacts.insert("Robert", "956-1745");

    // // Takes a reference and returns Option<&V>
    // match contacts.get(&"Daniel") {
    //     Some(&number) => println!("Calling Daniel: {}", call(number)),
    //     _ => println!("Don't have Daniel's number."),
    // }